### ISSUE TITLE
Console GraphiQL explorer stops working after tracking custom function

### ISSUE THREAD

**Issue by karibertils** [hasura/graphql-engine#6071]
I am using Hasura v1.3.2 and postgres 11 from [here](https://hub.docker.com/layers/bitnami/postgresql/11.8.0-debian-10-r76/images/sha256-986fb03adfe558bd6b427f26f56d8379f996bc335cfa2ee68e65b02e7158830c?context=explore)

The table

```
create table "user"
(
	id bigserial not null
		constraint user_pkey
			primary key,
	username text,
	password text
);
```

The function

```
CREATE OR REPLACE FUNCTION public.get_user()
RETURNS SETOF public."user" AS $$
  SELECT *
  FROM public."user"
  LIMIT 1
$$ LANGUAGE sql STABLE;
```

As soon as i press ```Track``` button for ```get_user``` in ```Untracked custom functions``` the GraphiQL explorer stops working. And instead shows ```NO SCHEMA AVAILABLE```

**Comment by kolharsam** [comment]
@karibertils this might be the case if you've used this custom function within a computed field as well. If this is indeed the case, then you could head over to #3757, where similar concerns are being tracked.

**Comment by karibertils** [comment]
Hm sounds very similar. But I'm not sure.

I did previously use computed field when I noticed this bug. Then removed the computed field and the SQL function.

Then tried creating new function and tracking it. This time the function takes no argument. Have not used this new function as computed field either. Tracking the function breaks the explorer and untracking fixes it again.

Did the first computed field trigger the bug then ?

**Comment by kolharsam** [comment]
@karibertils I have a feeling that you may have created functions that are polymorphic here and hence the bug persists. 

cc. @tirumaraiselvan

**Comment by karibertils** [comment]
@kolharsam 

Correct me if I'm wrong but this function doesn't look polymorphic to me ? It's the only function named get_user in the database.

![image](https://user-images.githubusercontent.com/5122320/97353641-321a3b00-188c-11eb-9599-fc90d5b89bab.png)
![image](https://user-images.githubusercontent.com/5122320/97353708-465e3800-188c-11eb-9c0e-7f794967de37.png)

If i untrack the function, the explorer works again, If i track again, it stops working. So very easy to reproduce.

**Comment by tirumaraiselvan** [comment]
Yes, there is a bug where if a function has a row type argument and it is tracked at top level (under query) and also as computed field, the introspection fails silently. 

We haven't really seen a real use-case where a function needs to have row type argument and also tracked at top -level. Usually, it is used as computed field. Can you describe if you need the function to be both top-level and computed field?

We should improve the experience here.

**Comment by karibertils** [comment]
Ahh,
No in my case i was planning to use a more complex function as a computed field when I noticed it was breaking after only tracking the function.

I created this dummy function just to test it out. I have no use for function taking row type argument in top-level query.

**Comment by b3j0f** [comment]
Is is possible to considerate computed fields such as tracked functions plz ?
